src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -12,100 +13,141 @@
 
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// Options used to load <see cref="SourceText"/>.
+    /// </summary>
+    public readonly record struct LoadTextOptions(SourceHashAlgorithm ChecksumAlgorithm);
```

Comment:
>public readonly record struct LoadTextOptions(SourceHashAlgorithm ChecksumAlgorithm); [](http://example.com/codeflow?start=4&length=85)

Are we really going to use everything that compiler is going to generate for this type? I recommend declaring a regular struct and explicitly declaring all members we want to have.

Reply:
We do you value equality.

Reply:
Sure, there are a few more things that it generates that are not strictly needed, but I don't see why I would need to manually write the rest just to avoid these.

Reply:
> but I don't see why I would need to manually write the rest just to avoid these.

Because we don't want to have these. This is not the first time we avoid using records in our implementation, especially in public API.